source/val/validate_atomics.cpp

```diff
@@ -213,7 +213,17 @@ spv_result_t AtomicsPass(ValidationState_t& _, const Instruction* inst) {
 
       // Then Shader rules
       if (_.HasCapability(SpvCapabilityShader)) {
-        if (storage_class == SpvStorageClassFunction) {
+        if (spvIsVulkanEnv(_.context()->target_env)) {
```

Comment:
Wasn't sure what WebGPU requires and the "correct" way spirv-val sorts out between enviornments